lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -98,9 +99,21 @@ def transform_block_filters(self, usage_info, block_structure):
                     if usage_key in library_children:
                         selected.append(selected_block)
 
+                #Get "canidate" blocks which have been manually selected
+                candidates = state_dict.get('candidates', [])
+
                 # Update selected
                 previous_count = len(selected)
-                block_keys = LibraryContentBlock.make_selection(selected, library_children, max_count, mode)
+                block_keys = LibraryContentBlock.make_selection(
+                    old_selected=list(map(tuple, selected)),
```

Comment:
```suggestion
                    # the cache stores `selected` as list[list[str]], but we want to work with list[tuple[str, str]]
                    old_selected=[(block_type, block_id) for block_type, block_id in selected],
```
nit: comprehensions are considered more pythonic. Also, here's a suggested comment.

Reply:
ack, pylint doesn't like this comprehension:

```
Unnecessary use of a comprehension, use list(self.selected) instead.
```

I'll put back what you had originally.